app/src/main/java/org/android/go/sopt/presentation/signup/SignUpViewModel.kt

```diff
@@ -6,49 +6,63 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import kotlinx.coroutines.launch
 import org.android.go.sopt.data.api.AuthFactory.ServicePool.authService
+import org.android.go.sopt.data.entity.User
 import org.android.go.sopt.data.model.remote.request.RequestPostSignUpDto
 import org.android.go.sopt.data.model.remote.response.ResponsePostSignUpDto
 import org.android.go.sopt.data.model.remote.response.base.BaseResponse
-import org.android.go.sopt.data.entity.User
 import org.android.go.sopt.util.PreferenceManager
 import org.android.go.sopt.util.code.CODE_DUPLICATED_ID
-import org.android.go.sopt.util.code.CODE_INCORRECT_INPUT
 import org.android.go.sopt.util.code.CODE_INVALID_INPUT
 import org.android.go.sopt.util.state.RemoteUiState
 import org.android.go.sopt.util.state.RemoteUiState.*
 import retrofit2.HttpException
 import timber.log.Timber
+import java.util.regex.Pattern
 
 class SignUpViewModel : ViewModel() {
-    private val preferenceManager = PreferenceManager()
-
     private val _signUpState = MutableLiveData<RemoteUiState>()
     val signUpState: LiveData<RemoteUiState>
         get() = _signUpState
 
-    // 양방향 데이터 바인딩을 위해 LiveData를 public으로 설정
+    // EditText에 입력된 값을 LiveData로 가져오는 양방향 데이터 바인딩을 위해
+    // LiveData를 public으로 설정했다.
     val _id = MutableLiveData("")
     private val id: String get() = _id.value?.trim() ?: ""
 
     val _pw = MutableLiveData("")
     private val pw: String get() = _pw.value?.trim() ?: ""
 
     val _name = MutableLiveData("")
-    val name: String get() = _name.value?.trim() ?: ""
+    private val name: String get() = _name.value?.trim() ?: ""
 
     val _hobby = MutableLiveData("")
-    val hobby: String get() = _hobby.value?.trim() ?: ""
-
-    private fun isValidInput(): Boolean {
-        return isValidId() && isValidPw() && name.isNotBlank() && hobby.isNotBlank()
-    }
+    private val hobby: String get() = _hobby.value?.trim() ?: ""
 
     fun isValidId(): Boolean {
-        return id.isNotBlank() && id.length in MIN_ID_LENGTH..MAX_ID_LENGTH
+        // 영문과 숫자를 적어도 하나씩 포함하는 6~10자리 문자열
+        val idRegex = """^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{6,10}$"""
+        val idPattern = Pattern.compile(idRegex)
+        return idPattern.matcher(id).matches()
     }
 
     fun isValidPw(): Boolean {
-        return pw.isNotBlank() && pw.length in MIN_PW_LENGTH..MAX_PW_LENGTH
+        // 영문, 숫자, 특수문자를 적어도 하나씩 포함하는 6~12자리 문자열
+        // 허용되는 특수 문자: !, @, #, $, %, ^, +, -, =
+        val pwRegex = """^(?=.*[A-Za-z])(?=.*\d)(?=.*[!@#$%^+\-=])[A-Za-z\d!@#$%^+\-=]{6,12}$"""
```

Comment:
```suggestion
        val pwRegex = """^(?=.*[A-Za-z])(?=.*\d)(?=.*[!@#$%^+\-=])[A-Za-z\d!@#$%^+\-=]{MIN_PW_LENGTH,MAX_PWD_LENGTH}$"""
```
정규식에 상수를 사용하는 것이 보일러플레이트를 줄일 수 있을 것 같습니다!